Feat: Add 2.10.0

diavrank opened the pull request:
[Changelog](https://github.com/meteor/meteor/blob/devel/docs/history.md#v2100-2023-01-13)

GeoffreyBooth commented:
The tests are failing for 2.10.0: https://github.com/disney/meteor-base/actions/runs/3954174678/jobs/6771345570

diavrank commented:
mmm it's weird, it says: FAIL: (no message displayed)

Do you know how can I show the error detail?

GeoffreyBooth commented:
You need to run the tests locally. Look inside `test.sh` to see what commands it runs and run them one-by-one on your local machine.

diavrank commented:
It´s crashing something in test.js 
<img width="1310" alt="image" src="https://user-images.githubusercontent.com/39581375/213883873-90b98efe-7aaf-44a0-b5aa-cd6f039465d8.png">

I will be checking it out this weekend and I will let you know if I could fix it

diavrank commented:
The problem is on line 9 of test.js, here:

````js

await page.goto('http://localhost/');
````

I added a catch statement, and this is the error:

````shell
➜  meteor-base git:(add-2100) ✗ ./test.sh 2.10.0            
Testing Docker image geoffreybooth/meteor-base:2.10.0...
Creating test app...
Building test app Docker image...
Launching test app...
Running test...
Before node test.js
error:  Error: net::ERR_CONNECTION_REFUSED at http://localhost/
    at navigate (file:///Users/diavrank/Documents/meteor-base/test/node_modules/puppeteer-core/lib/esm/puppeteer/common/Frame.js:210:23)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async Frame.goto (file:///Users/diavrank/Documents/meteor-base/test/node_modules/puppeteer-core/lib/esm/puppeteer/common/Frame.js:180:21)
    at async CDPPage.goto (file:///Users/diavrank/Documents/meteor-base/test/node_modules/puppeteer-core/lib/esm/puppeteer/common/Page.js:436:16)
    at async file:///Users/diavrank/Documents/meteor-base/test/test.js:9:3
FAIL: (no message displayed)
````

I already tried by updating puppeteer to the latest version but it didn't work. Also, I changed the port URL page to http://localhost:3000 but it didn't work either.

any suggestion @GeoffreyBooth ?

GeoffreyBooth commented:
> I already tried by updating puppeteer to the latest version but it didn’t work. Also, I changed the port URL page to http://localhost:3000 but it didn’t work either.

What this code is doing is launching the test app (like via the `meteor` command) and then running Puppeteer to connect to it and click some buttons and check for the expected HTML. Try running the app yourself locally via `meteor`; does it come up and you see the expected UI (something about “the button has been clicked X times” IIRC)? And then if you try to run it via the test Dockerfile, do you get the same? I think Puppeteer is connecting to port 80 because that’s what the Docker container’s port 3000 gets mapped to when it’s run.

Presumably the app isn’t launching correctly, so one or both of these direct ways of launching the test app should give you an error that you can investigate.

diavrank commented:
@GeoffreyBooth I already found the cause, it was because we had a conditional with regex for 2.1* version, so, it was never entering to 2.10 conditional. To fix it, I had to change it to 2.1.*  I guess I have to update the other one about 2.0*

<img width="733" alt="image" src="https://user-images.githubusercontent.com/39581375/213947355-f7220539-1b3f-4b93-9831-4f609b7f13a2.png">

however, as you can see, it's throwing another error. I guess is something related to a recent change you did in this part https://github.com/disney/meteor-base/commit/9dc41e38f03540e48954115c2d26bd72f048d7fe

diavrank commented:
Well, thinking twice, if 2.9.1 also had that error then I can guess that it's a normal behavior

````shell
➜  meteor-base git:(add-2100) ✗ ./test.sh 2.9.1 
Testing Docker image geoffreybooth/meteor-base:2.9.1...
Creating test app...
Building test app Docker image...
Launching test app...
Running test...
Before node test.js
Before elapsed variable
enter to else
PASS for geoffreybooth/meteor-base:2.9.1 with node:14.21.2-alpine after 5 min 14 sec
Error: No such image: example-app:latest

````

diavrank commented:
There was another error in CI:
<img width="608" alt="image" src="https://user-images.githubusercontent.com/39581375/213948303-5bcc2130-134e-4ff6-a19f-b316d683c72c.png">

I will be updating the correct version names of 2.1 and 2.0

diavrank commented:
mmm it seems 2.1.0 is not allowed:

````shell
➜  Documents meteor create myapp --release 2.1.0
Meteor 2.1.0: unknown release.
````

diavrank commented:
@GeoffreyBooth Fixed :)

GeoffreyBooth commented on test/test.js:
```diff
@@ -14,6 +14,8 @@ import puppeteer from 'puppeteer';
 
 		const element = await page.$('p');
 		message = await page.evaluate(element => element.textContent, element);
+	}catch(exception){
+		console.error('[test.js]: ', exception);
```
```suggestion
```

We don’t want to catch the exception, because we want the script to crash on error (in order to fail the test). I’m assuming this might’ve been left by accident while you were debugging.

diavrank replied:
Sure! Done https://github.com/disney/meteor-base/pull/134/commits/4ac72397d86604c3de7ad0149ecc655409dae322 . Thanks!

GeoffreyBooth commented:
> @GeoffreyBooth Fixed :)

Thank you for digging into this!
